staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -319,6 +319,7 @@ var newETCD3Client = func(c storagebackend.TransportConfig) (*clientv3.Client, e
 		tracingOpts := []otelgrpc.Option{
 			otelgrpc.WithPropagators(tracing.Propagators()),
 			otelgrpc.WithTracerProvider(c.TracerProvider),
+			otelgrpc.WithMessageEvents(otelgrpc.ReceivedEvents, otelgrpc.SentEvents),
```

Comment:
do we need to add this everywhere we create tracing options (I see three changes in this PR but we have ~7 places we create WithTracerProvider in kube code)?

does this apply to otelhttp as well or just otelgrpc?

Reply:
This applies only to otelgrpc. Most instrumentation libraries (e.g. otelhttp, otelgrpc, etc.) allow you to provide a non-global TracerProvider, since that is a best practice.  The WithMessageEvents option is otelgrpc-specific, and keeps the current behavior.